src/umessage/umessagebuilder.rs

```diff
@@ -387,14 +450,16 @@ impl<'a> UMessageBuilder<'a> {
     /// let method_to_invoke = UUri::try_from("my-vehicle/cabin/1/rpc.doors")?;
     /// let reply_to_address = UUri::try_from("my-cloud/dashboard/1/rpc.response")?;
     /// let token = String::from("this-is-my-token");
-    /// let message = UMessageBuilder::request(&method_to_invoke, &reply_to_address, &uuid_builder.build(), 5000)
-    ///                     .with_token(&token)
-    ///                     .build_with_payload(&uuid_builder, "lock".into(), UPayloadFormat::UPAYLOAD_FORMAT_TEXT)?;
+    /// let message = UMessageBuilder::request(method_to_invoke, reply_to_address, 5000)
+    ///                     .with_message_id(uuid_builder.build())
+    ///                     .with_token(token.clone())
+    ///                     .build_with_payload("lock".into(), UPayloadFormat::UPAYLOAD_FORMAT_TEXT)?;
     /// assert_eq!(message.attributes.token, Some(token));
     /// # Ok(())
     /// # }
     /// ```
-    pub fn with_token(&'a mut self, token: &'a String) -> &'a mut UMessageBuilder {
+    pub fn with_token(&mut self, token: String) -> &mut UMessageBuilder {
+        assert!(self.message_type == UMessageType::UMESSAGE_TYPE_REQUEST);
```

Comment:
I guess the reason you use assert instead of using `Result` is to make the code cleaner, right?

Reply:
Returning a `Result` would be different from the value returned by all the other setters and thus prevent fluent API usage. Also, I guess in this case it would be ok to panic because it would be a programming error to include a token in a message other than a request.